app/services/sessions/clear_old_sessions_service.rb

```diff
@@ -0,0 +1,42 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+module Sessions
+  class ClearOldSessionsService
+    class << self
+      ##
+      # Drop all sessions for the given user
+      def call!(days_ago: 30)
+        # sessions expire after 30 days of inactivity by default
+        expiration_time = Time.zone.today - days_ago.days
+
+        sessions_table = ActiveRecord::SessionStore::Session.table_name
+        ActiveRecord::Base.connection.execute "DELETE FROM #{sessions_table} WHERE updated_at < '#{expiration_time}'"
```

Comment:
Why are we dropping to a raw SQL Query? We use the model to determine the table name, we use AR::Base connection so rails is fully loaded here, so we can also just do

```suggestion
        ActiveRecord::SessionStore::Session.where("updated_at < ?", days_ago.days.ago).delete_all
```